diff --git a/cmake/module/FindANTLR.cmake b/cmake/modules/FindANTLR.cmake

--- a/cmake/modules/FindANTLR.cmake
+++ b/cmake/modules/FindANTLR.cmake
@@ -12,17 +12,16 @@ find_package(Java QUIET COMPONENTS Runtime)
 # ANTLR JAR HANDLING
 if(NOT ANTLR_JAR_LOCATION)
   find_file(ANTLR_JAR_LOCATION
-            NAMES antlr-4.12.0-complete.jar antlr-4.13.0-complete.jar antlr-4.13.1-complete.jar antlr.jar antlr4.jar antlr-4.jar 
-            HINTS /usr /usr/local /usr/local/share /usr/local/share/java /usr/local/lib/ /usr/share /usr/share/java /usr/local/Homebrew ~/homebrew /usr/local/homebrew/Cellar /opt/homebrew/Cellar
-            PATH_SUFFIXES antlr 4.12.0 4.13.0 4.13.1 antlr/4.12.0 antlr/4.13.0 antlr/4.13.1 
+            NAMES antlr4 antlr
+            HINTS /usr/bin
             NO_CMAKE_SYSTEM_PATH
             NO_SYSTEM_ENVIRONMENT_PATH)
     message("ANTLR Jar: ${ANTLR_JAR_LOCATION}")
 endif()
 
-if(ANTLR_JAR_LOCATION AND Java_JAVA_EXECUTABLE)
+if(ANTLR_JAR_LOCATION)
   execute_process(
-      COMMAND ${Java_JAVA_EXECUTABLE} -jar ${ANTLR_JAR_LOCATION}
+      COMMAND ${ANTLR_JAR_LOCATION}
       OUTPUT_VARIABLE ANTLR_COMMAND_OUTPUT
       ERROR_VARIABLE ANTLR_COMMAND_ERROR
       RESULT_VARIABLE ANTLR_COMMAND_RESULT
